sdcm/nemesis.py

```diff
@@ -4318,6 +4319,119 @@ def disrupt_add_remove_mv(self):
+                    raise NodeSetupFailed(node=new_node, error_msg=log_messages[0])
+
+                if new_node.db_up() and new_node.jmx_up():
+                    self.log.info("Node %s bootstrapped succesfull", new_node.name)
+            except Exception as exc:  # pylint: disable=broad-except
+                self.log.warning("Node bootstrap was aborted with error: %s", exc)
+            finally:
+                self.log.debug("Stop node bootstrap process")
+                bootstrap_process.join(60)
+                bootstrap_process.terminate()
+                self.log.debug("Node bootstrap process stopped")
+
+        trigger = partial(
+            start_bootstrap, new_node=new_node)
+
+        terminate_patterns = self.target_node.raft.get_log_pattern_for_bootstrap_abort()
+
+        self.use_nemesis_seed()
+        terminate_pattern = random.choice(terminate_patterns)
+        self.log.info("Stop bootsrap process after log message: '%s'", terminate_pattern)
+
+        log_follower = new_node.follow_system_log(patterns=[terminate_pattern])
+
+        watcher = partial(
+            self._call_disrupt_func_after_expression_logged,
+            log_follower=log_follower,
+            disrupt_func=new_node.stop_scylla,
+            disrupt_func_kwargs={},
+            delay=0
+        )
+
+        with EventsSeverityChangerFilter(new_severity=Severity.WARNING,
+                                         event_class=DatabaseLogEvent,
+                                         regex=".*init - Startup failed.*",
+                                         extra_time_to_expiration=30), \
+                adaptive_timeout(operation=Operations.NEW_NODE, node=self.target_node, timeout=3600) as bootstrap_timeout:
+            ParallelObject(objects=[trigger, watcher], timeout=bootstrap_timeout).call_objects(ignore_exceptions=True)
+
+        host_id_searcher = new_node.follow_system_log(patterns=['Setting local host id to'])
+
+        def clean_unbootstrapped_node():
+            new_node_host_ids = []
+            if found_stings := list(host_id_searcher):
+                for line in found_stings:
+                    new_node_host_id = line.split(" ")[-1].strip()
+                    self.log.info("Node %s has host id: %s in log", new_node.name, new_node_host_id)
+                    new_node_host_ids.append(new_node_host_id)
+            self.log.debug("New host was not properly bootstrapped. Terminate it")
+            self._terminate_cluster_node(new_node)
+            self.log.info("Sleep for gossiper updater")
+            time.sleep(60)
+            self.target_node.raft.clean_group0_garbage(True)
+            if new_node_host_ids:
+                for host_id in new_node_host_ids:
+                    self.target_node.run_nodetool(
+                        f"removenode {host_id}", ignore_status=True, retry=3, warning_event_on_exception=True)
+
+            assert self.target_node.raft.is_group0_tokenring_num_of_nodes_consistent(), \
```

Comment:
so we interrupt the test in case of assertion fail - shouldn't we clean after that?